File: app/src/main/java/fr/nspu/dev/recordvoicelearning/utils/OrderPeerEnum.java
package fr.nspu.dev.recordvoicelearning.utils;

public enum OrderPeerEnum{
        KNOWLEDGE_ASCENDING(0),
        KNOWLEDGE_DESCENDING(1);

        Integer order = 0;

        OrderPeerEnum(int i) {
            this.order = i;
        }

        public int toInt(){
            return order;
        }

        public static OrderPeerEnum toOrderPeerEnum(Integer id){
            switch (id){
                case 0:
                    return KNOWLEDGE_ASCENDING;
                case 1:
                    return KNOWLEDGE_DESCENDING;
                default:
                    //return default value
                    return KNOWLEDGE_ASCENDING;
            }
        }
}

File: app/src/main/java/fr/nspu/dev/recordvoicelearning/utils/OrderPeerEnum.kt
package fr.nspu.dev.recordvoicelearning.utils

enum class OrderPeerEnum private constructor(i: Int) {
    KNOWLEDGE_ASCENDING(0),
    KNOWLEDGE_DESCENDING(1);

    internal var order: Int? = 0

    init {
        this.order = i
    }

    fun toInt(): Int = order!!

    companion object {
        fun toOrderPeerEnum(id: Int?): OrderPeerEnum {
            when (id) {
                0 -> return KNOWLEDGE_ASCENDING
                1 -> return KNOWLEDGE_DESCENDING
                else -> return KNOWLEDGE_ASCENDING
            }
        }
    }
}